### ISSUE TITLE
API to access Zotero DB

### ISSUE THREAD

**Issue by stefanopagliari** [aidenlx/obsidian-zotero#82]
Thank you for this great plugin. The integration with the Zotero DB in the Zotero Plugin works great. As mentioned in [Discord](https://discord.com/channels/686053708261228577/722584061087842365/1036051647845122190) it would be helpful (at least for me but I'm sure also for others) if you considered introducing an API or some other mechanism to allow other plugins to read the Zotero DB. Thanks

**Comment by stefanopagliari** [comment]
@aidenlx sorry to bother you about this. I was wondering if you could provide any suggestion on how it could be possible to read the Zotero DB from another plugin. My attempts to integrate sqlite into my plugin have hit a wall

**Comment by aidenlx** [comment]
Sorry for the long wait! The next major update obsidian plugin will arrive in two week or so. I've been busy with the work and refactoring the plugin. Can you first try describe the needs for the API? 

> could provide any suggestion on how it could be possible to read the Zotero DB from another plugin

I'm using a [custom build of better-sqlite3](https://github.com/aidenlx/better-sqlite3/), and download binaries for the github release based on machine's architecture. I've put database instance inside worker so that reading db won't block UI. The core logic for init database instance can be found here, Although I still recommend you to use API exposed from this plugin in next version: 

https://github.com/aidenlx/obsidian-zotero/blob/dev/lib/db-worker/src/database.ts

**Comment by stefanopagliari** [comment]
@aidenlx @MSBack 
Thank you for this.
Sorry for not seeing before your message from February 20.
Ideally I would like a way for other plugins to be able to run searches on the Zotero DB once your plugin is installed and activated.

**Comment by aidenlx** [comment]
currently available APIs is listed in here: https://obzt.aidenlx.top/reference/api/interfaces/PluginAPI

**Comment by FeralFlora** [comment]
Hey @aidenlx, is the Zotlit API still available? The documentation link above is dead, and I can't find anything about it in the new docs. I am interested in using the API to develop a plugin that provides some Zotero related workflows.

**Comment by aidenlx** [comment]
Yes, it's still there, it's just that I haven't migrate that part to new docs. I'll see if API doc generation is still feasible in new document framework.

**Comment by FeralFlora** [comment]
> Yes, it's still there, it's just that I haven't migrate that part to new docs. I'll see if API doc generation is still feasible in new document framework.

Will you give an update here when the API docs are back?

**Comment by FeralFlora** [comment]
Bump

**Comment by FeralFlora** [comment]
Any update on the Zotlit API?
